jlm/hls/backend/rvsdg2rhls/decouple-mem-state.cpp

```diff
@@ -84,10 +84,11 @@ trace_edge(
       state_edge = get_mem_state_user(out);
       continue;
     }
-    auto si = util::AssertedCast<rvsdg::SimpleInput>(state_edge);
-    auto sn = si->node();
-    auto new_si = util::AssertedCast<rvsdg::SimpleInput>(new_edge_user);
-    auto new_sn = new_si->node();
+
+    auto si = state_edge;
+    auto sn = rvsdg::TryGetOwnerNode<rvsdg::SimpleNode>(*si);
+    auto new_si = new_edge_user;
+    auto new_sn = rvsdg::TryGetOwnerNode<rvsdg::SimpleNode>(*new_si);
```

Comment:
This `sn` and `new_sn` should use `AssertGetOwnerNode`, no?

Reply:
Fixed